common/build_data_linkstamp.h

```diff
@@ -0,0 +1,25 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_COMMON_BUILD_DATA_LINKSTAMP_H_
+#define CARBON_COMMON_BUILD_DATA_LINKSTAMP_H_
+
+namespace Carbon::BuildData::Internal {
+
+// See build_data.h; the list of names here should match.
+//
+// These are exposed as non-constexpr so that `build_data_linkstamp.cpp` can be
+// compiled at the end, even though dependencies are added earlier.
+//
+// Also, when build_data_linkstamp.cpp is compiled, this doesn't receive deps,
+// so we can't use things like `llvm::StringRef` here. As a result, we use
+// `build_data.h` as an intermediary to do a `StringRef` wrap.
+extern const char platform[];
+extern const bool build_coverage_enabled;
+extern const char target_name[];
+extern const char build_target[];
```

Comment:
Could we use `std::string_view` for these? That would let us compute the size in the `.cpp` file, and access that size when initializing our `llvm::StringRef` public variables. I think that would avoid needing to do `strlen` to compute the size in a dynamic initializer.

Reply:
To check, do you have thoughts on just having them exposed as std::string_view instead of llvm::StringRef?

Reply:
I find mixing `string_view` and `StringRef` a bit annoying, and we have a *lot* of other `StringRef`, so I liked keeping things consistent with the wrappers you have. It also seems nice to add a decoupling in case the constraints on linkstamp change more in the future.

Reply:
Done

Reply:
Switching back:#5464